### PR TITLE
docs(LinuxCommand): add linux command grep

### LINKED ISSUE #41: [enhancement]add 'grep' to LinuxCommand
## Is your feature request related to a problem? Please describe

I found that a most commonly-used command 'grep' has not been included in the doc.

## Description of the new feature/enhancement

The whole explanation of 'grep' will be added, maybe along with a few samples of how to write regular expression. 
However, I am not sure how much content of regular expression should be conveyed. Perhaps I only need to explain about '*','.'and'[]'?
In addition, a contrast of 'grep' and 'find' will be added, following the style of the contrast between 'more' and 'less'.

### PR DESCRIPTION
**Pull request by nagi2330**
## Description

a complete introduction to the command 'grep' is added to doc

- [x] add basic usage format of command grep
- [x] add arguments of command grep
- [x] add considerations of command grep
- [x] add typical examples of command grep
- [x] add descriptions of command grep

Resolves: #41 
## How Has This Been Tested?

grep.md has passed local mdl test according to .markdownlint.json as rules

## Types of changes

- Documentation (non-breaking change which updates documentation)

## Checklist:

- [x] Code follows the code style of this project.
- [x] Changes follow the **CONTRIBUTING** guidelines.
- [x] Update necessary documentation accordingly.
- [x] Lint and tests pass locally with the changes.
- [x] Check issues and pull requests first. You don't want to duplicate effort.

### CODE REVIEW

**Comment by nagi2330** [comment]
@BobAnkh I am curious what is wrong with my pr and commit mesage. gitmagic pointed that there is sth. wrong with the titles but i cannot find.  some help is needed! onegai~

**Comment by BobAnkh** [comment]
You missed a space after `:` :joy:

**Comment by nagi2330** [comment]
got it! that is so tricky😂

**Comment by BobAnkh** [comment]
Still facing space problem :joy:

Don’t underestimate this space, it’s a small but unignored part to help us get style consistency

**Comment by nagi2330** [comment]
@BobAnkh  finished fixing typo and suggestion in yesterday's review has been adopted！

File: docs/LinuxCommand/grep.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+作为Linux最重要的文本处理工具之一的grep命令，其功能非常强大，需要对其有一定的掌握。由于grep命令主要是进行搜索匹配，所以为了更好的使用grep命令，最好能对正则表达式和通配符这两种常见的指定待匹配内容的方式有所了解，限于篇幅只在后文给出一些例子。
+
+在使用grep时，对于二进制文件进行处理要谨慎使用'-a'参数，使用之后由于将二进制文件作为文本文件处理，很可能在输出中会产生难以预测的内容，如果此时将输出定向到命令行，这些输出有可能被系统理解为命令，会产生糟糕的结果。当然，使用'-a'可以在不知编码形式的文件中找到更多可能存在的匹配部分，尽管结果是不安全的。
+
+## 5. 常用形式
+
+查找file.txt文件中所有含有'object'的行
+
+```bash
```
Comment (BobAnkh) [inline docs/LinuxCommand/grep.md]: Change all `bash` to `console`

File: docs/LinuxCommand/grep.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+与find命令相比，grep主要用于对文件内容进行搜索，find命令则是对文件名进行搜索。
+
+作为Linux最重要的文本处理工具之一的grep命令，其功能非常强大，需要对其有一定的掌握。由于grep命令主要是进行搜索匹配，所以为了更好的使用grep命令，最好能对正则表达式和通配符这两种常见的指定待匹配内容的方式有所了解，限于篇幅只在后文给出一些例子。
+
+在使用grep时，对于二进制文件进行处理要谨慎使用'-a'参数，使用之后由于将二进制文件作为文本文件处理，很可能在输出中会产生难以预测的内容，如果此时将输出定向到命令行，这些输出有可能被系统理解为命令，会产生糟糕的结果。当然，使用'-a'可以在不知编码形式的文件中找到更多可能存在的匹配部分，尽管结果是不安全的。
+
+## 5. 常用形式
+
+查找file.txt文件中所有含有'object'的行
```
Comment (BobAnkh) [inline docs/LinuxCommand/grep.md:117]: Better to refactor this section.

Create a file and tell readers what's in it and show them the result of your command in the example

You can just creat one file but use it repeatedly in every example

File: docs/LinuxCommand/grep.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+
+```bash
+grep 'object' file.txt
+```
+
+查找file.txt文件中不含'object'的行且只打印满足条件的行数
+
+```bash
+grep -vc 'object' file.txt
```
Comment (BobAnkh) [inline docs/LinuxCommand/grep.md]: Should give example of `grep -v` command

**Review by BobAnkh — CHANGES_REQUESTED** [review]
Great job! But still some things need to be changed

**Comment by nagi2330** [comment]
my fault，i push the wrong version again😂i will correct this very soon

**Comment by BobAnkh** [comment]
To be honest, as a first-timer, you actually did much greater than others :smile: Keep on moving!

**Comment by BobAnkh** [comment]
@nagi2330

If you are ready for review and resolve the problem I proposed, please reply to the review and resolve conversation, then comment to me

**Comment by nagi2330** [comment]
> If you are ready for review and resolve the problem I proposed, please reply to the review and resolve conversation, then comment to me

I think it is time for reviewing! :D

**Comment by nagi2330** [comment]
> If you are ready for review and resolve the problem I proposed, please reply to the review and resolve conversation, then comment to me

I think it is time for another reviewing! :D

**Comment by BobAnkh** [comment]
Ok I will review it and please next time reply to the reveiw before resolve it to make sure you don't resolve it unintentionally

**Review by BobAnkh — APPROVED** [review]
You are doing a great job!

I will merge it soon.

**Comment by nagi2330** [comment]
> Great job! But still some things need to be changed
replying to the review is doing something like this,is it?

**Comment by nagi2330** [comment]
> You are doing a great job!
> 
> I will merge it soon.

thanks! I will turn to investigating tool tree soon.

**Comment by BobAnkh** [comment]
You should reply to that issue and then I will assign it to you

**Pull request merged**